src/adjoint/masterRoutines.F90

```diff
@@ -607,6 +609,56 @@ subroutine master_b(wbar, xbar, extraBar, forcesBar, dwBar, nState, famLists, &
     real(kind=realType) :: dummyReal, dummyReald
     logical ::resetToRans
 
+    ! you may be thinking we do we need this when we always do a forward pass before solving the
+    ! adjoint? well it is because we run this code multiple times to setup the adjoint. On each
+    ! reverse pass the varibles (namely the halo cell states) can be changed so it is import to
+    ! reset them with another forward pass.
+
+      do sps=1,nTimeIntervalsSpectral
+         do nn=1,nDom
+           call setPointers(nn, 1, sps)
+
+            ! Compute the pressures/viscositites
+            call computePressureSimple(.False.)
+  
+            ! Compute Laminar/eddy viscosity if required
+            call computeLamViscosity(.False.)
+            call computeEddyViscosity(.False.)
+  
+            ! Make sure to call the turb BC's first incase we need to
+            ! correct for K
+            if (equations == RANSequations) then
+               call BCTurbTreatment
+               call applyAllTurbBCthisblock(.True.)
+            end if
+            call applyAllBC_block(.True.)
+
+
+            call timeStep_block(.True.)
```

Comment:
@camader this is the bug fix (line 637) I referenced in the meeting.
In addition to resetting the boundary conditions the spectral radius needs to be calculated since it isn't calculated in the residual calculation. 

But I'm a bit confused why the spectral radius isn't set by calling the getResiduals() routine on the python level

Reply:
It looks like it is an issue with using the blockettes

Reply:
can you explain the bug here as well?

Reply:
After you instantiate an ADflow instance with a restart file.
If you call 
`CFDSolver.getResiduals(ap)
CFDSolver.computeJacobianVectorProductBwd(....)`
you get the wrong answer

But if you call 
`CFDSolver.getResiduals(ap)
CFDSolver.computeJacobianVectorProductFwd(....)`
CFDSolver.computeJacobianVectorProductBwd(....)`
you get the right answer

If you trace the error back you see that this is because the spectral radii (radI, radJ, radK) aren't set by getResiduals() with the default options

I just learned that if you set the option `useblockettes' to False there is no issue. So it seems like the spectral radii are being zeroed before use in master_b when using blockettes.

Per neil's suggestion I'm going to make a separate issue to discuss this further